beetsplug/edit.py

```diff
@@ -262,10 +291,20 @@ def edit_objects(self, objs, fields):
                         return False
 
                 # Show the changes.
+                # If the objects are not on the DB yet, we need a copy of their
+                # original state for show_model_changes.
+                if all(not obj.id for obj in objs):
```

Comment:
OK! Once again, thank you for investigating and discarding my half-baked ideas. :smiley: This solution looks very straightforward.